src/Registration/cReg/NiftiImageData3DBSpline.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+SyneRBI Synergistic Image Reconstruction Framework (SIRF)
+Copyright 2017 - 2020 University College London
+
+This is software developed for the Collaborative Computational
+Project in Synergistic Reconstruction for Biomedical Imaging (formerly CCP PETMR)
+(http://www.ccpsynerbi.ac.uk/).
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+/*!
+\file
+\ingroup Registration
+\brief Class for deformation field transformations.
+
+\author Richard Brown
+\author SyneRBI
+*/
+
+#include "sirf/Reg/NiftiImageData3DBSpline.h"
+#include "sirf/Reg/NiftiImageData3DDeformation.h"
+#include "sirf/NiftyMoMo/BSplineTransformation.h"
+
+using namespace sirf;
+
+template<class dataType>
+NiftiImageData3DBSpline<dataType>::NiftiImageData3DBSpline(const NiftiImageData3DDeformation<dataType> &def, const float spacing[])
+{
+    // not marked const, so copy
+    float spacing_nonconst[3] = {spacing[0], spacing[1], spacing[2]};
+    // Get any of the tensor components as a 3d image
+    auto ref_sptr = def.get_tensor_component(0);
```

Comment:
when moving this to the converter, this will have to be the ref of the CPG I believe

Reply:
Now I use `_template_ref_sptr` for both the `forward` and `backward` methods.